### ISSUE TITLE
Numpy > 2.0.0 changes output of decode for CategoricalHMM

### ISSUE THREAD

**Issue by Lblackhurst29** [hmmlearn/hmmlearn#558]
## I am reporting a bug

Numpy == 2.1.0
hmmlearn == 0.3.2
python == 13.11.3

Using a pretrained categorical HMM to decode a tutorial dataset causes a different result to usual. Rather than a mix of all states it only decoded it to one of the four.

Changing the version of numpy to before their new version (1.24.6) recovers expected result. No error is thrown when using the newer version of numpy, just an unexpected result.

**Comment by Takahisa-Ishikawa** [comment]
I have got a simillar issue using `GaussianHMM.predict()`

Below is the code to reproduce issue.

```py
import numpy as np
from hmmlearn import hmm
import hmmlearn

# model
n_states = 3
model = hmm.GaussianHMM(n_components=n_states, covariance_type="diag", n_iter=1000)

# parameter
transmat = np.array([[0.7, 0.2, 0.1],
                     [0.3, 0.5, 0.2],
                     [0.3, 0.3, 0.4]])

startprob = np.array([0.6, 0.3, 0.1])

# set parameter
model.startprob_ = startprob
model.transmat_ = transmat

# sensor data
np.random.seed(42)
n_samples = 300
X = np.random.rand(n_samples, 3)  # 3軸の加速度データ（x, y, z）

# fit model
model.fit(X)

# predict
hidden_states = model.predict(X)

# output
print("numpy version: ", np.__version__)
print("hmmlearn version: ", hmmlearn.__version__)
print("Estimated hidden states:")
print(hidden_states)
```

Different versions of numpy produce different output.

correct
```sh
numpy version:  1.24.4
hmmlearn version:  0.3.2
Estimated hidden states:
[2 0 1 0 0 1 0 1 2 2 0 2 0 0 1 0 0 2 2 1 0 0 1 2 1 2 0 0 0 2 1 2 0 1 1 1 1
 1 2 1 2 1 2 0 1 1 2 1 1 0 0 2 2 0 0 0 1 1 0 2 0 0 2 1 2 2 2 1 0 0 2 1 2 2
 1 1 2 2 1 1 2 0 2 0 0 2 0 0 2 0 2 0 2 2 0 1 1 1 1 1 1 2 0 1 1 0 1 2 2 0 2
 1 2 1 1 2 2 2 1 2 2 2 2 1 1 2 0 2 2 1 0 0 0 0 2 0 0 2 0 2 2 0 2 1 0 0 0 0
 2 0 2 0 1 1 2 0 2 0 2 2 2 2 1 0 1 1 2 0 0 2 0 0 2 1 2 0 1 2 2 1 2 2 1 2 2
 0 2 1 1 2 2 2 1 1 0 1 1 0 0 1 1 1 1 0 0 0 2 1 0 0 2 0 0 0 1 2 0 2 0 2 2 1
 1 1 1 0 2 1 2 1 1 1 1 2 0 0 2 0 2 0 0 2 1 1 2 0 0 2 1 2 1 1 2 0 1 2 2 0 0
 2 0 0 0 0 2 2 2 1 2 0 2 2 0 2 1 0 0 0 0 2 1 0 1 0 1 1 0 0 2 1 0 0 2 1 1 1
 1 2 0 0]
```

wrong
```sh
numpy version:  2.0.1
hmmlearn version:  0.3.2
Estimated hidden states:
[1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1]
```

**Comment by anntzer** [comment]
Fixed by releasing hmmlearn 0.3.3.
